### PR TITLE
Navigation: Landing pages behind feature toggles

### PR DESCRIPTION
**Pull request by ashharrison90**
**What this PR does / why we need it**:

- adds new components `NavLandingPage`/`NavLandingPageCard`
  - these create generic, fairly basic looking landing pages (we can improve this later!) for a given `navId`
- adds landing page routes for alerting (`/alerting`) and config (`/admin`)
- adds a special landing page route for each plugin 

note: this is all behind the feature toggle 

**Which issue(s) this PR fixes**:

For https://github.com/grafana/grafana/issues/50791

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ashharrison90** [comment]
@torkelo would be good to sync up at some point and discuss this, particularly around plugins as there's quite a bit of overlap with the stuff you're working on i think 😅

File: public/app/core/components/AppChrome/NavLandingPage.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  content: css({
+    display: 'flex',
+    flexDirection: 'column',
+    gap: theme.spacing(2),
+  }),
+  grid: css({
+    display: 'grid',
+    gap: theme.spacing(2),
+    gridTemplateColumns: 'repeat(auto-fill, minmax(300px, 1fr))',
+    gridAutoRows: 'minmax(200px, 1fr)',
```
Comment (joshhunt) [inline public/app/core/components/AppChrome/NavLandingPage.tsx:73]: this just makes each row a minimum of 200px high?

File: public/app/routes/routes.tsx [hunk 2]
```diff
@@ -179,6 +180,14 @@ export function getAppRoutes(): RouteDescriptor[] {
...
       ),
     },
+    {
+      path: '/apps',
+      component: () => <NavLandingPage navId="apps" />,
+    },
+    {
+      path: '/a/:pluginId/landing-page',
+      component: (props) => <NavLandingPage navId={`plugin-page-${props.match.params.pluginId}`} />,
+    },
```
Comment (joshhunt) [inline public/app/routes/routes.tsx:190]: should this route maybe be feature-flagged?

**Review by joshhunt — APPROVED** [review]
Not a big fan of the routing for it - i think the home/landing page for apps should be at the root of it's URL (e.g. `/a/:pluginId`), but as long as it's behind a feature flag we can iterate from here 😌

File: public/app/core/components/AppChrome/NavLandingPage.tsx (hunk 1 above)
Comment (ashharrison90) [reply public/app/core/components/AppChrome/NavLandingPage.tsx:73]: yeah actually this just really needs to be `gridAutoRows: '200px'`, i'll change 👍

**Comment by ashharrison90** [comment]
@joshhunt yeah the routing is tricky. right now i don't think there's anything stopping a plugin from putting a page on the base route, and i don't want to override that route if something does exist there. a plugin could also be putting something on the `/landing-page` route right now, it's just less likely. need to work out the best way to achieve this. i'll feature toggle the route for now anyway whilst we figure out the best way forward 👍

**Comment by joshhunt** [comment]
I figure if an app has a page at `/a/:pluginId` then that should be the home/landing page?

File: public/app/routes/utils.ts [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
 export function isSoloRoute(path: string): boolean {
   return /(d-solo|dashboard-solo)/.test(path?.toLowerCase());
 }
+
+export function pluginHasRootPage(pluginId: string, navTree: NavLinkDTO[]): boolean {
+  return Boolean(
+    navTree
+      .find((navLink) => navLink.id === 'apps')
+      ?.children?.find((app) => app.id === `plugin-page-${pluginId}`)
+      ?.children?.some((page) => page.url?.endsWith(`/a/${pluginId}/`))
```
Comment (torkelo) [inline public/app/routes/utils.ts:12]: don't think we should enforce that last forward slash

**Review by torkelo — APPROVED** [review]
This looks great!

File: pkg/api/index.go [hunk 4]
```diff
@@ -298,6 +303,11 @@ func (hs *HTTPServer) getNavTree(c *models.ReqContext, hasEditPerm bool, prefs *
 			SortWeight: dtos.WeightConfig,
 			Children:   configNodes,
 		}
+		if hs.Features.IsEnabled(featuremgmt.FlagTopnav) {
+			configNode.Url = "/admin"
```
[the remark this answers is not in the document]
Comment (ashharrison90) [reply pkg/api/index.go:307]: yeah agreed. gonna tackle that in a follow-up PR... not sure what behaviour's best. we could:

- give server admin it's own "landing page" as well
- remove the duplicate breadcrumb

idk, i'll play around

**Pull request merged**